File: examples/python/m3-agent-loop/policy.py
"""M3：最小 allow-list Policy。

只回答"这个工具名允许执行吗"，不做参数级策略、速率限制或审批流——那些是
后续里程碑的问题。
"""

from __future__ import annotations

from typing import Iterable


class AllowListPolicy:
    """按工具名的白名单。"""

    def __init__(self, allowed: Iterable[str]) -> None:
        self._allowed = set(allowed)

    def check(self, tool_name: str) -> bool:
        """工具名是否在白名单中。"""

        return tool_name in self._allowed

File: examples/rust/m3-agent-loop/src/policy.rs
//! M3: minimal allow-list policy.
//!
//! Independent of `policy_engine::AllowListPolicy` (that one implements
//! `agent_core::PolicyEvaluator`, tied to `AgentState`). M3's loop only
//! needs "is this tool ID allowed."

use std::collections::BTreeSet;

pub struct AllowListPolicy {
    allowed: BTreeSet<String>,
}

impl AllowListPolicy {
    pub fn new<I, S>(allowed: I) -> Self
    where
        I: IntoIterator<Item = S>,
        S: Into<String>,
    {
        Self {
            allowed: allowed.into_iter().map(Into::into).collect(),
        }
    }

    pub fn check(&self, tool_id: &str) -> bool {
        self.allowed.contains(tool_id)
    }
}
